apps/mn-synt-arbeidsforhold-service/src/main/java/no/nav/registre/testnorge/mn/syntarbeidsforholdservice/service/ArbeidsforholdHistorikkService.java

```diff
@@ -0,0 +1,179 @@
+package no.nav.registre.testnorge.mn.syntarbeidsforholdservice.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+
+import no.nav.registre.testnorge.mn.syntarbeidsforholdservice.consumer.SyntrestConsumer;
+import no.nav.registre.testnorge.mn.syntarbeidsforholdservice.domain.Arbeidsforhold;
+import no.nav.registre.testnorge.mn.syntarbeidsforholdservice.domain.ArbeidsforholdMap;
+import no.nav.registre.testnorge.mn.syntarbeidsforholdservice.domain.Opplysningspliktig;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ArbeidsforholdHistorikkService {
+    private final Executor executor;
+    private final SyntrestConsumer syntrestConsumer;
+    private final IdentService identService;
+    private final OpplysningspliktigService opplysningspliktigService;
+    private final Random random = new Random();
+
+
+    private Map<LocalDate, Arbeidsforhold> createArbeidsforholdHistorikk(final String ident, Iterator<LocalDate> dates) {
+        var map = new HashMap<LocalDate, Arbeidsforhold>();
+        var startdato = dates.next();
+        log.info("Generer for {} den {}.", ident, startdato);
+        var first = syntrestConsumer.getFirstArbeidsforhold(startdato, ident, null);
+        map.put(startdato, first);
+
+        if (!dates.hasNext()) {
+            return map;
+        }
+
+        var previous = first;
+        var historikk = syntrestConsumer.getArbeidsforholdHistorikk(previous, startdato).iterator();
+
+        while (dates.hasNext()) {
+            LocalDate kalendermaaned = dates.next();
+            log.info("Generer for {} den {}.", ident, kalendermaaned);
```

Comment:
"den" gir mening for en dato, men når det er en kalendermåned bak passer det kanskje bedre med "i måned "?

Reply:
kalendermaaned har formatet 2020-11 så vet ikke hva jeg skulle klat denne